### PR TITLE
Fixes #216 remove \Exception type hint

### LINKED ISSUE #216: Argument 1 passed to ZendDeveloperTools\Exception\SerializableException::__construct() must be an instance of Exception
The phenomenon of #10 (supposedly fixed since 2012) happens again. I'm running the current master branch, it started to appear some weeks ago.

Any time the application generates an exception, the toolbar blocks it and only shows

    Fatal error: Uncaught TypeError: Argument 1 passed to ZendDeveloperTools\Exception\SerializableException::__construct() must be an instance of Exception, instance of Error given, called in /…/vendor/zendframework/zend-developer-tools/src/Collector/ExceptionCollector.php on line 45 and defined in /…/vendor/zendframework/zend-developer-tools/src/Exception/SerializableException.php:26 Stack trace:
    #0 /…/vendor/zendframework/zend-developer-tools/src/Collector/ExceptionCollector.php(45): ZendDeveloperTools\Exception\SerializableException->__construct(Object(Error))
    #1 /…/vendor/zendframework/zend-developer-tools/src/Profiler.php(210): ZendDeveloperTools\Collector\ExceptionCollector->collect(Object(Zend\Mvc\MvcEvent))
    #2 /…/vendor/zendframework/zend-developer-tools/src/Listener/ProfilerListener.php(93): ZendDeveloperTools\Profiler->collect(Object(Zend\Mvc\MvcEvent))
    #3 [ in /…/vendor/zendframework/zend-developer-tools/src/Exception/SerializableException.php on line 26

### PR DESCRIPTION
**Pull request by samsonasik**

### CODE REVIEW

**Comment by Ocramius** [comment]
@samsonasik can you verify it with a test? Doing something like following should suffice:

```php
try {
    (new \stdClass)->iDoNotExist();
} catch (\Throwable $exception) { 
    $serializable = new SerializableException($exception);
   // assert here
}

// fail if no $exception thrown
```

**Comment by samsonasik** [comment]
@Ocramius `(new \stdClass)->iDoNotExist();` will result `Error` in php7

**Comment by Ocramius** [comment]
It will be a throwable instance

**Comment by samsonasik** [comment]
@Ocramius I got this error:

```php
...PHP Fatal error:  Uncaught TypeError: Argument 1 passed to ZendDeveloperTools\Exception\SerializableException::__construct() must be an instance of Exception, instance of Error given,
```

**Comment by Ocramius** [comment]
@samsonasik https://3v4l.org/gS75T

**Comment by samsonasik** [comment]
@Ocramius done ;), tests added, should be ok now ;)

File: test/Exception/SerializableExceptionTest.php [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+            $this->assertEquals('Division by zero', $serializable->getMessage());
+
+            return;
+        } catch (\Exception $exception) {
+            $serializable = new SerializableException($exception);
+            $this->assertEquals('Division by zero', $serializable->getMessage());
+        }
+    }
+
+    public function testStdClassCallNotExistMethod()
```
Comment (Ocramius) [inline test/Exception/SerializableExceptionTest.php]: Test requires PHP 7 - you can use the PHPUnit annotation for that

File: test/Exception/SerializableExceptionTest.php [hunk 2]
```diff
@@ -0,0 +1,38 @@
+<?php
+
+namespace ZendDeveloperToolsTest\Exception;
+
+use PHPUnit_Framework_TestCase;
+use ZendDeveloperTools\Exception\SerializableException;
+
+class SerializableExceptionTest extends PHPUnit_Framework_TestCase
+{
+    public function testExceptionAndThrowable()
```
Comment (Ocramius) [inline test/Exception/SerializableExceptionTest.php]: Can simplify to just an exception test case (instantiate a new exception)

**Comment by samsonasik** [comment]
@Ocramius tests updated. Travis now green \m/

**Comment by snapshotpl** [comment]
Anyone can merge it? It fail on my php7 dev machine ;(

**Comment by DannyvdSluijs** [comment]
I would also like this to be merged.

**Comment by ikovalyov** [comment]
+1 please merge

**Comment by vladas** [comment]
@samsonasik Is there any reason you removed the type hinting? Exception and Error are instances of Throwable. In my mind it should have been enough to replace \Exception with \Throwable.

**Comment by samsonasik** [comment]
\Throwable only works with php7, we use it in php ^5.6

**Comment by DannyvdSluijs** [comment]
@vladas \Throwable is only availible from PHP7. It would break any compatibility with 5.X

**Comment by vladas** [comment]
Ok, fair enough.

**Comment by MatthiasKuehneEllerhold** [comment]
Any news on this PR?

**Comment by weierophinney** [comment]
Thanks, @samsonasik

**Pull request merged**